pkg/ruler/ruler.go

```diff
@@ -97,7 +106,67 @@ type Ruler struct {
 
 	// Per-user notifiers with separate queues.
 	notifiersMtx sync.Mutex
-	notifiers    map[string]*notifier.Notifier
+	notifiers    map[string]*rulerNotifier
+}
+
+type rulerNotifier struct {
```

Comment:
Can you please add a comment for this explaining what it is and why we want it?

Reply:
Done.